XDMA/linux-kernel/xdma/alinx_arch.c

```diff
@@ -93,6 +93,58 @@ u32 alinx_get_tx_drop_packets(struct pci_dev *pdev) {
         return priv->total_tx_drop_count;
 }
 
+u64 alinx_get_normal_timeout_packets(struct pci_dev *pdev) {
+        struct xdma_dev* xdev = xdev_find_by_pdev(pdev);
+        struct xdma_private* priv = netdev_priv(xdev->ndev);
+        u32 regval = read32(xdev->bar[0] + REG_NORMAL_TIMEOUT_COUNT);
+
+        if (regval < (u32)(priv->last_normal_timeout & 0xFFFFFFFF)) {
+                // Overflow
+                priv->last_normal_timeout += 0x100000000;
+        }
+        priv->last_normal_timeout &= 0xFFFFFFFF00000000;
+        priv->last_normal_timeout |= (u64)regval;
```

Comment:
This could handle overflow more cleanly
```suggestion
        u32 diff = regval - ((u32)priv->last_normal_timeout); // This could handle overflow
        priv->last_normal_timeout += (u64)diff;
```

For example (In this example, I use single octet)
_regval_ goes 0x30 → 0x40:
diff = 0x10.
0x30(last) + 0x10(diff) ⇒ 0x40

_regval_ goes 0xff → 0x02:
diff = 0x03 (0x02 - 0xff is -0xfd, But `diff` is unsigned, So it is 0x03)
0xff(last) + 0x03(diff) ⇒ 0x0102

Reply:
Also, this used on many location. How about make it into a function?
eg: `add_u32_counter(*u64 sum, u32 value)`